### PR TITLE
We stopped using allow_internal_unstable a while ago

### PR DESCRIPTION
**Pull request by oli-obk**
cc @rust-lang/wg-const-eval

### CODE REVIEW

File: src/stability.md [hunk 1]
```diff
@@ -89,24 +89,28 @@ You can see an example of stabilizing a feature at [#75132](https://github.com/r
...
+attribute that allows the given features to be used in stable macros.
+
+## rustc_allow_const_fn_unstable
+
+`const fn`, while not directly exposing their body to the world, are going to get
+evaluated at compile time in stable crates. If their body does something const-unstable,
+that could lock us into certain features indefinitely by accident. Thus no unstable const
+features are allowed inside stable `const fn`. Sometimes we do know that a feature will get
+stabilized, just not when or there is a stable (but e.g. runtime-slow) workaraound, so we
+could always fall back to some stable version if we scrapped the unstable feature.
```
Comment (RalfJung) [inline src/stability.md:104]:
```suggestion
features are allowed inside stable `const fn`.

However, sometimes we do know that a feature will get
stabilized, just not when or there is a stable (but e.g. runtime-slow) workaraound, so we
could always fall back to some stable version if we scrapped the unstable feature.
In those cases, the rustc_allow_const_fn_unstable attribute can be used to allow some
unstable features in the body of a stable `const fn`.
```

File: src/stability.md [hunk 2]
```diff
@@ -89,24 +89,32 @@ You can see an example of stabilizing a feature at [#75132](https://github.com/r
...
+
+## rustc_allow_const_fn_unstable
+
+`const fn`, while not directly exposing their body to the world, are going to get
+evaluated at compile time in stable crates. If their body does something const-unstable,
+that could lock us into certain features indefinitely by accident. Thus no unstable const
+features are allowed inside stable `const fn`.
+
+However, sometimes we do know that a feature will get
+stabilized, just not when or there is a stable (but e.g. runtime-slow) workaraound, so we
```
Comment (JohnTitor) [inline src/stability.md:105]:
```suggestion
stabilized, just not when or there is a stable (but e.g. runtime-slow) workaround, so we
```

File: src/stability.md [hunk 3]
```diff
@@ -89,24 +89,32 @@ You can see an example of stabilizing a feature at [#75132](https://github.com/r
...
+You also need to take care to uphold the `const fn` invariant that calling it at runtime and
+compile-time needs to behave the same (see also [this blog post][blog]). This means that you
+may not create a `const fn` that e.g. transmutes a memory address to an integer,
 because the addresses of things are nondeterministic and often unknown at
 compile-time.
 
-Always ping @oli-obk, @RalfJung, and @Centril if you are adding more
-`allow_internal_unstable` attributes to any `const fn`
+Always ping @rust-lang/wg-const-eval if you are adding more
+`rustc_allow_const_fn_unstable` attributes to any `const fn`
```
Comment (JohnTitor) [inline src/stability.md:117]: nit:

```suggestion
`rustc_allow_const_fn_unstable` attributes to any `const fn`.
```

File: src/stability.md [hunk 4]
```diff
@@ -89,24 +89,32 @@ You can see an example of stabilizing a feature at [#75132](https://github.com/r
...
+
+## rustc_allow_const_fn_unstable
+
+`const fn`, while not directly exposing their body to the world, are going to get
+evaluated at compile time in stable crates. If their body does something const-unstable,
+that could lock us into certain features indefinitely by accident. Thus no unstable const
+features are allowed inside stable `const fn`.
+
+However, sometimes we do know that a feature will get
+stabilized, just not when or there is a stable (but e.g. runtime-slow) workaround, so we
```
Comment (camelid) [inline src/stability.md:105]: I think a comma might make this a little clearer?

```suggestion
stabilized, just not when, or there is a stable (but e.g. runtime-slow) workaround, so we
```

**Comment by JohnTitor** [comment]
@RalfJung Is this good to merge?

**Comment by RalfJung** [comment]
Yes. :)

**Review by JohnTitor — APPROVED** [review]
(Rubber-stamping)

**Comment by JohnTitor** [comment]
Thanks!

**Pull request merged**